octobot_services/service_feeds/abstract_service_feed.py

```diff
@@ -0,0 +1,148 @@
+#  Drakkar-Software OctoBot-Services
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+from abc import *
+import threading
+
+from octobot_channels.channels.channel import get_chan, set_chan
+
+from octobot_commons.asyncio_tools import run_coroutine_in_asyncio_loop
+
+
+# ****** Unique dispatcher side ******
+from octobot_services.abstract_service_user import AbstractServiceUser
+from octobot_services.channel.abstract_service_feed import AbstractServiceFeedChannelProducer
+
+
+class AbstractServiceFeed(AbstractServiceUser, threading.Thread, AbstractServiceFeedChannelProducer):
```

Comment:
With this inheritance it won't be able to be cythonized. I don't know if you wanted to cythonize it ?

Reply:
(If you don't cythonize it, please type your methods :wink: )

Reply:
threading.Thread has been removed (now in tentacles)
For AbstractServiceFeedChannelProducer it's more complicated, it's better to keep this inheritance to avoid code duplication.
With some tweaks on AbstractServiceUser, we can avoid this inheritance, it will complicate the code though. Do you think it's worth cythonizing ? The main processing is being done in implementations, here are mostly init procedures

Reply:
No I don't think it's worth cythonizing here.